ViewModel/BrowserViewModel.swift

```diff
@@ -227,6 +227,26 @@ final class BrowserViewModel: NSObject, ObservableObject,
         decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
     ) {
         guard let url = navigationAction.request.url else { decisionHandler(.cancel); return }
+        
```

Comment:
Thank you for this, it is indeed working for macOS.
There are a couple of things, that should be addressed:
- This class is re-used for iOS as well, and for that reason this code needs to be within a ``#if os(macOS)`` block, as at the moment it won't compile for iOS.
- We do not use `;` in our codebase, for early returns after `guard` statements.
Theoretically you could do it on one line, like so:
```
... else {
    return decisionHandler(.cancel)
}
```
This is still confusing, and taking advantage of the callback function return type.
The preferred and more explicit way is eg:
```
... else {
    decisionHandler(.cancel)
    return
}
```
Please use this format.

- another issue is that most of this "new tab opening" code has been copy pasted from another function, while the return type is somewhat different, I think, the "new tab creation" can be moved to a new private function, perhaps that could return `Bool`. The advantage of this will be that the code is in 1 place, and the function itself can be wrapped into `#if os(macOS)`.

Reply:
@BPerlakiH, I have updated it by adding a condition for MacOS and moved the code for creating a new tab into a private function.